documentation/upgrade/argo-checkout-0.8.0-to-0.9.0.md

```diff
@@ -0,0 +1,42 @@
+# Upgrading from `@shopify/argo-checkout(-react)` 0.8.x to 0.9.x
+
+In `@shopify/argo-checkout` and `@shopify/argo-checkout-react` version 0.9.x, significant changes were made to the build process to enable smaller bundle sizes.
+
+In particular, projects using React (through the `@shopify/argo-checkout-react` package) are about [75% smaller](https://github.com/Shopify/argo-checkout/pull/63) for the template extension. This was done by replacing all usage of React with [`@remote-ui/mini-react`](https://github.com/Shopify/remote-ui/tree/main/packages/mini-react), a tiny re-implementation of the core React API.
+
+In order to get access to these improvements, you **must** also upgrade your `@shopify/argo-run` `devDependency`, to at least version `0.3.0`, which contains the build configuration that performs this replacement.
+
+For most projects, this change should be completely transparent, with one exception: you can now remove the `React` default import from all your components, as this new build system uses the [`jsx-runtime` transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) to remove this bit of boilerplate:
```

Comment:
Is this still optional to remove? I think it's.

Reply:
It is optional, but there is a performance benefit to removing it (noted below)